File: AD-HOC/2378.js
let [leituras, capacidade] = lines.shift().split(' ').map(Number);
let total = 0;
let encontrou = false;

for (let i = 0; i < leituras; i++) {
  let [S, E] = lines.shift().split(' ').map(Number);

  total -= S;
  total += E;

  if (total > capacidade) {
    encontrou = true;
    console.log("S");
    break;
  }
}

if (!encontrou) {
  console.log("N");
}

File: AD-HOC/2378.py
entrada = list(map(int, input().split()))
leituras = entrada[0]
capacidade = entrada[1]
total = 0
encontrou = False

for i in range(leituras):
  entradas = list(map(int, input().split()))
  S = entradas[0]
  E = entradas[1]

  total -= S
  total += E

  if (total > capacidade):
    encontrou = True
    print("S")
    break

if (encontrou == False):
  print("N")
